setup.py

```diff
@@ -53,25 +51,24 @@
     package_dir={"": "src"},
     include_package_data=True,
     zip_safe=False,
-    python_requires=">=2.7",
+    python_requires=">=3.8",
     install_requires=[
```

Comment:
There are circular dependency errors. I don't know if something could be done in this package to solve the problem. Maybe reduce the number of dependencies?

Reply:
I have added there the list of dependencies that `tox -e dependencies` reports...

Reply:
The problem seems to be `plone.api`.
@mauritsvanrees  @jensens @gforcada can't an addon depend on `plone.api`?

Reply:
There's indeed a circular dependency here, this is what `tox -e dependencies` reports:

```
Detecting cyclic edges
found edge products-cmfplone -> plone-api
found edge products-cmfplone -> plone-app-discussion
found edge plone-app-discussion -> plone-api
Cyclic dependencies detected

```

Reply:
I have seen this before.  With the versions of Plone 6.0 this should not happen, nor with 6.1.  I wonder if the constraints are not active and you simply get the latest releases.  That could explain it.

It would be nice if the circular dependencychecker would have an option to only complain about a cyclic dependency that includes the currently developed package.

Certainly for an add-on it is perfectly normal to depend on `plone.api`.

Reply:
- if you depend on plone.api you can skip to depend on Products.CMFPlone.
- plone.app.discussion should be a soft dependency since it is a core addon not available by default since Plone 6.1.

Reply:
> I wonder if the constraints are not active and you simply get the latest releases. That could explain it.

@mauritsvanrees It uses 6.0-dev:

https://github.com/plone/plone.formwidget.recaptcha/blob/erral-updates/tox.ini#L203

> It would be nice if the circular dependencychecker would have an option to only complain about a cyclic dependency that includes the currently developed package.

It seems that this error is only shown when the package depends on `plone.api`. But anyway, shouldn't this circular dependency be fixed in Plone? Or is this not a circular dependency from Plone's point of view?

Reply:
> if you depend on plone.api you can skip to depend on Products.CMFPlone.

@jensens the problem is that if you use `plone.api` in the package and don't declare it in `install_requires`, [z3c.dependencychecker](https://github.com/plone/plone.formwidget.recaptcha/blob/erral-updates/tox.ini#L69-L77) complains. Maybe we should have an entry for `Products.CMFPlone` in `pyproject.toml` like this:

https://github.com/plone/plone.formwidget.recaptcha/blob/erral-updates/pyproject.toml#L84

> plone.app.discussion should be a soft dependency since it is a core addon not available by default since Plone 6.1.

Plone 6.0 is used here.

Reply:
Actually, I would argue that there is no cyclic dependency:

```
Detecting cyclic edges
found edge products-cmfplone -> plone-api
found edge products-cmfplone -> plone-app-discussion
found edge plone-app-discussion -> plone-api
Cyclic dependencies detected
```

You have:

* cmfplone -> api
* cmfplone -> discussion -> api

This just means there are two routes from cmfplone to api, one short and one long.  There is no cycle back to cmfplone.

@jensens Problem in `pipdeptree` or `pipforester` maybe?

Reply:
@mauritsvanrees In Plone 6.0, `Products.CMFPlone` depends on [plone.api](https://github.com/plone/Products.CMFPlone/blob/1cc9cb0f2b2cd0b09b6330b04c772c3473c971d2/setup.py#L56) and `plone.api` depends on [Products.CMFPlone](https://github.com/plone/plone.api/blob/47245aa8d3a0bde5c645818a441dac44e1919a0f/setup.py#L34). Isn't this a cyclic dependency?

Reply:
Ah, right.  But that is not the cycle that is reported by `pipforester`, so it is a bit strange.  At least that is not how I read the report.

It would be nice if someone would implement this [`pipforester` feature request](https://github.com/collective/pipforester/issues/6).  That could solve it.

Or solve the cyclic dependency, but that is done already: problem is that this is done in Plone 6.1, and we can't really do it in 6.0.

Reply:
It seems that `pipforester` only complains if the current package depends on two packages that have circular dependencies on each other. This might make sense. We would only have to depend on one. In this case, we could depend only on `Products.CMFPlone` and configure `z3c.dependencychecker` to not complain about the lack of dependency on `plone.api`:

```
[tool.dependencychecker]
Products.CMFPlone = [ 'plone.api' ]
```

Reply:
> problem is that this is done in Plone 6.1, and we can't really do it in 6.0.

@mauritsvanrees  why can't we fix this in Plone 6? Would this be a breaking change?